### PR TITLE
Fix and test for list of ceilometer metrics

### LINKED ISSUE #89: Ceilometer telemetry unable to parse multiple items in a payload
If multiple instances are created in a project, the Ceilometer parsing logic is unable to deal with it.

### PR DESCRIPTION
**Pull request by pleimer**
Fixes #89 

This patch adjusts ceilometer metric parsing logic to handle metrics coming from multiple instances at once in a list object. Unit tests have been re factored as well and a python script included to help ease the process of creating correct output data for said unit tests.

I am working on testing this in STF right now and will post the results once that completes.

### CODE REVIEW

**Comment by pleimer** [comment]
Here are the results of querying Prometheus with this update. The OSP providing the data contains two running instances.

```json
{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": [
      {
        "metric": {
          "__name__": "ceilometer_cpu",
          "counter": "cpu",
          "cpu": "375b2194-efc4-4674-b595-70315f00edcb",
          "endpoint": "prom-http",
          "project": "e56191ef77744c599dbcecae6af176bb",
          "publisher": "telemetry.publisher.controller-0.redhat.local",
          "resource": "375b2194-efc4-4674-b595-70315f00edcb",
          "service": "stf-default-ceilometer-telemetry-smartgateway",
          "type": "cumulative",
          "unit": "ns"
        },
        "values": [
          [
            1600183344,
            "5512150000000"
          ]
        ]
      },
      {
        "metric": {
          "__name__": "ceilometer_cpu",
          "counter": "cpu",
          "cpu": "d8bd99b6-6fd8-4c02-a2e3-efbf596df636",
          "endpoint": "prom-http",
          "project": "e56191ef77744c599dbcecae6af176bb",
          "publisher": "telemetry.publisher.controller-0.redhat.local",
          "resource": "d8bd99b6-6fd8-4c02-a2e3-efbf596df636",
          "service": "stf-default-ceilometer-telemetry-smartgateway",
          "type": "cumulative",
          "unit": "ns"
        },
        "values": [
          [
            1600183344,
            "466290000000"
          ]
        ]
      }
    ]
  }
}
```

**Comment by leifmadsen** [comment]
Looks pretty good to me, and I really appreciate seeing so much testing being added.

I'm going to defer to those more skilled in Golang to provide you a review though.

File: internal/pkg/metrics/incoming/ceilometer.go [hunk 1]
```diff
@@ -107,26 +107,42 @@ func (c *CeilometerMetric) sanitize(data string) string {
...
+		dP.Publisher = message["publisher_id"].(string)
+		dP.Payload = pl.(map[string]interface{})
+
+		dP.DataSource.SetFromString("ceilometer")
+		dP.SetNew(true)
+		dP.SetData(&dP)
+		dataPoints = append(dataPoints, &dP)
+	}
+
+	return dataPoints, nil
```
Comment (csibbitt) [inline internal/pkg/metrics/incoming/ceilometer.go:145]: I was able to meaningfully review up to this point, and the testing looks sane to me, but I'm not familiar enough with the ceilometer formats or how to test it to really validate that the meat of this is good. IMO if it appears to solve the problem we started out with (no metrics when there are multiple instances), then it's good enough for me! :)

**Review by csibbitt — APPROVED** [review]
LGTM, but I did not test it myself.

**Pull request merged**